src/hotspot/cpu/ppc/stubGenerator_ppc.cpp

```diff
@@ -3544,6 +3544,442 @@ class StubGenerator: public StubCodeGenerator {
+
+        // Note that skip_xxsel is indexed because this code is contained
+        // in a C++ loop (the emitted code in this unroll loop doesn't
+        // loop).  The indexing allows the creation of a unique labels for
+        // each iteration of the unrolled loop.
+        __ bind(skip_xxsel[unroll_cnt]);
+
+        // Use the lower_nibble to select a mask "M" from the lookup table.
+        __ xxperm(M, maskLUT, lower_nibble);
+
+        // "bit" is used to isolate which of the bits in M is relevant.
+        __ xxperm(bit, bitposLUT, higher_nibble->to_vsr());
+
+        // Each element of non_match correspond to one each of the 16 input
+        // characters.  Those elements that become 0x00 after the xxland
+        // instuction are invalid Base64 characters.
+        __ xxland(non_match->to_vsr(), M, bit);
+
+        // Compare each element to zero
+        //
+        // vmcmpequb_ sets the EQ bit of CCR6 if no elements compare equal.
+        // Any element comparing equal to zero means there is an error in
+        // that element.  Note that the comparison result register
+        // non_match is not referenced again.  Only CCR6-EQ matters.
+        __ vcmpequb_(non_match, non_match, vec_0s);
+        __ bne_predict_not_taken(CCR6, unrolled_loop_exit);
+
+        // The Base64 characters had no errors, so add the offsets
+        __ vaddubm(input, input, offsets);
+
+        // Pack
+        //
+        // Legend for the tables below: b0, b1, .. b15 are the bytes of
+        // decoded binary data.  The specifier after the colon depicts
+        // which bits are there.  The bit numbering is big endian style
+        // (bit 0 is the most significant).  The || is a concatenate
+        // operator (same terminology as used in the Power ISA 3.x
+        // document).  Strings of 0's are a field of zeros with the shown
+        // length.
+
+        if (PowerArchitecturePPC64 >= 10) {
+            // Note that only e15..e8 are shown here because the extract
+            // bit pattern is the same in e7..e0.
+            //
+            // +===============+=============+======================+======================+=============+=============+======================+======================+=============+
+            // |    Vector     |     e15     |         e14          |         e13          |     e12     |     e11     |         e10          |          e9          |     e8      |
+            // |    Element    |             |                      |                      |             |             |                      |                      |             |
+            // +===============+=============+======================+======================+=============+=============+======================+======================+=============+
+            // | after vaddubm | 00||b0:0..5 | 00||b0:6..7||b1:0..3 | 00||b1:4..7||b2:0..1 | 00||b2:2..7 | 00||b3:0..5 | 00||b3:6..7||b4:0..3 | 00||b4:4..7||b5:0..1 | 00||b5:2..7 |
```

Comment:
An extra line here showing how the 8 6-bit values above get mapping into 6 bytes greatly help my brain out. (likewise for the <P10 case below too)

Reply:
Just to make sure I understand, you're not asking for a change here, is that right?

Reply:
I think the first line should also express the initial layout of the 6 bit values similar to the linked algo.  I think changing this comment add an extra line which describes the bits as they leave `vaddubm` would be helpful to understand the demangling here. (e.g the `00aaaaaa 00bbbbbb 00ccccc 00dddddd` comments in the linked paper)

Reply:
I think I have addressed the issues in this comment with the latest commits.  Reversing the order of the bytes in the tables seems to make the tables easier to understand, and also make the vector constant declarations consistent: all use the ARRAY_TO_LXV_ORDER macro now.

The 00aaaaa (etc.) bit fields are added to the tables..  I'm not 100% sure they help much, but at least the comments follow the original paper in a clearer way.